MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -126,14 +126,14 @@
 		.shaderInt8 = true,
 		.descriptorIndexing = _vulkan12FeaturesNoExt.descriptorIndexing,
 		.shaderInputAttachmentArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
-		.shaderUniformTexelBufferArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
```

Comment:
When does `_metalFeatures.arrayOfTextures` fail? Are you running CTS on an ancient devices or OS?

Perhaps we should disable the `VK_EXT_descriptor_indexing` extension completely in these cases. Perhaps test `_metalFeatures.arrayOfTextures` in `MVKPhysicalDevice::initExtensions()`.

Reply:
I had thought that we were enabling `shaderSampledImageDynamicIndexing` unconditionally, but, on closer inspection, we weren't. I've changed it to disable the extension if the requisite Metal features aren't supported, as you suggested.